### ISSUE TITLE
[BUG] Logging does not work again with pennylane

### ISSUE THREAD

**Issue by qlan3** [PennyLaneAI/pennylane#2006]
### Expected behavior

Similar to this [issue](https://github.com/PennyLaneAI/pennylane/issues/207), I was expecting writing logs into a log file by setting:
```
logging.basicConfig(
      format='%(asctime)s - %(levelname)s: %(message)s',
      filename='log.txt',
      filemode='w'
)
```

### Actual behavior

However, the logger still outputs everything to stdout. I guess that PennyLane may overwrite my logging setting.

### Source code

```shell
# Some qml functions I used
qml.BasisStatePreparation
qml.StronglyEntanglingLayers
qml.AngleEmbedding
qml.StronglyEntanglingLayers
qml.expval
qml.PauliZ
qml.StronglyEntanglingLayers.shape
qml.device
qml.QNode
qml.qnn.TorchLayer
```

### System information

```shell
Name: PennyLane
Version: 0.19.1
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: None
Author-email: None
License: Apache License 2.0
Location: /home/qlan3/miniconda3/lib/python3.9/site-packages
Requires: scipy, semantic-version, autograd, pennylane-lightning, cachetools, networkx, autoray, toml, numpy, appdirs
Required-by: PennyLane-Lightning
Platform info:           Linux-5.8.0-44-generic-x86_64-with-glibc2.31
Python version:          3.9.5
Numpy version:           1.20.3
Scipy version:           1.7.1
Installed devices:
- default.gaussian (PennyLane-0.19.1)
- default.mixed (PennyLane-0.19.1)
- default.qubit (PennyLane-0.19.1)
- default.qubit.autograd (PennyLane-0.19.1)
- default.qubit.jax (PennyLane-0.19.1)
- default.qubit.tf (PennyLane-0.19.1)
- default.qubit.torch (PennyLane-0.19.1)
- default.tensor (PennyLane-0.19.1)
- default.tensor.tf (PennyLane-0.19.1)
- lightning.qubit (PennyLane-Lightning-0.19.0)
```

### 

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by CatalinaAlbornoz** [comment]
Hi @qlan3, thank you for reporting this bug! We will look into it.

**Comment by raghavven** [comment]
I got the same error, tried the following code (when uncommented), it created the log file , but did not write anything to it nor it printed on the jupyter notebook. When commented and ran the same code, it displayed the debug messages below the cell. Need to print to the log file that is being created. Also, to note, the log file created only when I used 'force=True', not sure why though. 

import logging

logging.basicConfig(level=logging.DEBUG, format = "{asctime} {levelname:<8} {message}",style='{',
                    #filename='trylog.log',
                    #filemode='w',force=True
                   )
logger = logging.getLogger()

Thank you

**Comment by josh146** [comment]
Thanks for reporting this @raghavven and @qlan3! I searched the PL codebase, and it seems that the only place PennyLane is importing the `logging` module is

https://github.com/PennyLaneAI/pennylane/blob/01d014093c73cd9459c33fe2bb935d0cef3d10c4/pennylane/configuration.py#L19-L25

I wonder if removing this fixes the issue.

One final question @raghavven and @qlan3: which order were you importing the logger? Before or after importing PennyLane?

**Comment by qlan3** [comment]
Thank you @josh146 ! I removed line 20 and 25, replace line 58 with a `print` function. Now my logger works well but I get a raised FileNotFoundError error with msg "No PennyLane configuration file found." However, this won't affect my usage for now.

I imported the logger and PennyLane from different files while PennyLane should be imported first.

**Comment by josh146** [comment]
Thanks for checking this @qlan3! The issue must be with the `log.getLogger()` call in the `configuration.py` module then --- I think if called first, before your custom logger configuration, it must be overriding it.

**Comment by qlan3** [comment]
Actually, I tried to reconfigure `logging.basicConfig` with my own configuration right before every call of `logger.info`. However, this didn't work either. I'm not familiar with package `logging`. Maybe this is the problem from `logging` not `PennyLane`?

**Comment by raghavven** [comment]
Hi @josh146 . I import Pennylane at the very first cell. 'import logging' comes much later. Also, I tried the logging again and go the following thing written the log file created.  Below are the codes I used to create login file and the stuff it got written into it.

1) Logging code used
logging.basicConfig(level=logging.DEBUG, format = "{asctime} {levelname:<8} {message}",style='{',
                    filename='trylog_13212021H3Plus.log',
                    filemode='w',force=True
                   )
logger = logging.getLogger()

2) Stuff written to the trylog_13212021H3Plus.log file
2021-12-13 12:11:53,566 DEBUG    Creating converter from 7 to 5
2021-12-13 12:11:53,566 DEBUG    Creating converter from 5 to 7
2021-12-13 12:11:53,567 DEBUG    Creating converter from 7 to 5
2021-12-13 12:11:53,567 DEBUG    Creating converter from 5 to 7
1970-01-05 18:19:32,289 DEBUG    Creating converter from 5 to 3

Additionally, I take this opportunity to note that I am looking for details of molecular orbital coefficients, charges, etc. to be written to my logging file, which is basically my motivation to using logging. I want to get a log file with details similar to what one would get classical computational chemistry while using packages such as Gaussian or PySCF. I am not sure how to achieve it. Can you kindly throw some light on that as well. 

Thank you.

**Comment by josh146** [comment]
Thanks @raghavven! In your case, does setting up your logger _prior_ to importing PennyLane temporarily fix the issue? 

> I want to get a log file with details similar to what one would get classical computational chemistry while using packages such as Gaussian or PySCF. 

I don't have much experience with computational chemistry logging, but @soranjh might have some insights 🙂

**Comment by raghavven** [comment]
Thanks @josh146 . Updating the most recent tries. 
In order to be clear with logging , I re ran the while thing . case 1) import pennylane first and then logging , case 2) importing logging (very first cell) first and pennylane. Logging level was, DEBUG ,both the cases created logging file and wrote things tot it. When I changed the level from DEBUG to INFO , didnt write anything to the log file. Guess it could be a thing of VQE and not logging issue.  Also, thanks for tagging @soranjh. I would like to know how to print results and other observables that can be seen in usual Quantum Chemistry packages. Regarding, logging, guess it is settled for now at least. Thanks a lot for your help.

**Comment by soranjh** [comment]
Hi @raghavven. 

When you use pennylane-qchem, the Hartre-Fock calculations needed to build the molecular Hamiltonian are performed with OpenFermion-PySCF/PsiPy. In the case of PyScf, the molecular information is written in the hdf5 file and information such as molecular orbital coefficients can be directly extracted from that. See for example:

```
import pennylane as qml
from pennylane import qchem
from pennylane import numpy as np
import openfermion

symbols = ["H", "H"]
geometry = np.array([0.0, 0.0, 0.0,
                     0.0, 0.0, 1.4])

H, qubits = qml.qchem.molecular_hamiltonian(symbols, geometry)

molecule = openfermion.MolecularData(filename='./pyscf/sto-3g/molecule.hdf5')
molecule.canonical_orbitals
```

If you are using the differentiable Hartree-Fock [solver](https://pennylane.readthedocs.io/en/stable/code/qml_hf.html) in the qml.hf module, then the information such as atomic and molecular orbitals can be directly obtained from the molecule class:

**Comment by raghavven** [comment]
Thank you @soranjh . That really helped.

**Comment by Jaybsoni** [comment]
@raghavven and @qlan3, 

I just want to confirm if there are any action items here? From the above thread it seems you were able to get your loggers working by swapping the order of the imports and/or removing the line from `configuration.py` which called the logger in the first place? 

Thanks again for pointing this out! Let me know if you are seeing any additional issues around this bug and I can address this it in a bug fix PR, otherwise if it was just a matter of keeping an eye on which order the logger is being imported then we can close the issue here 👍🏼 

Cheers,

**Comment by Jaybsoni** [comment]
So as a closing comment I would like summarize: 

When using logging and PennyLane, one should make sure to import and setup logging first BEFORE importing PL to ensure the logger settings are not ignored. 

```python
import logging
logging.basicConfig(filename='mylogfile.txt', level=logging.INFO)
logging.info('A log entry.')

import pennylane as qml
```

@josh146, after looking into this, it seems that the logging we currently do is not used anywhere else in the codebase. In fact, we don't log to file at all. It might be worth removing completely. Let me know what you think.

**Comment by leofang** [comment]
Sorry, late to the party. But the import order should not matter. It's a trap too easy to fall into for end users. I offered a two-liner fix in #3731 and explained this bug happens.
